Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/storage/FamilyTestDatabase.kt

```diff
@@ -99,7 +99,17 @@ interface FamilyCoronaTestDao {
     suspend fun update(identifier: TestIdentifier, update: (FamilyCoronaTest) -> FamilyCoronaTest) {
         get(identifier)?.let {
             val updated = update(it.test).toEntity()
-            insert(updated)
+            if (it != updated) insert(updated)
+        }
+    }
+
+    @Transaction
+    suspend fun update(updates: List<Pair<TestIdentifier, (FamilyCoronaTest) -> FamilyCoronaTest>>) {
+        updates.forEach {
```

Comment:
maybe you could introduce `insertAll` and use DAO to do that ?

Reply:
I introduced it here https://github.com/corona-warn-app/cwa-app-android/pull/5029/files#diff-3996b0ad1c7eec53fe025846ded86fdfd68df2bf965747a7ec62019f906170eeR113-R116